helix-core/src/syntax.rs

```diff
@@ -968,14 +982,19 @@ impl Syntax {
     // TODO: Folding
 }
 
+const LAYER_MODIFIED: u32 = 0b001;
+const LAYER_MOVED: u32 = 0b010;
+const LAYER_TOUCHED: u32 = 0b100;
+
 #[derive(Debug)]
 pub struct LanguageLayer {
     // mode
     // grammar
     pub config: Arc<HighlightConfiguration>,
     pub(crate) tree: Option<Tree>,
     pub ranges: Vec<Range>,
-    pub depth: usize,
+    pub depth: u32,
+    pub flags: u32,
```

Comment:
We already use the bitflags crate in other crates, maybe use that here?

Reply:
I check this with cargo tree and it somehow didn't show up but now looking at the Cargo.lock by hand its there. Will update this to use the bitflags crate

Reply:
PR has been updated to use bitflags